autometa/config/environ.py

```diff
@@ -269,36 +272,26 @@ def get_versions(program=None):
         `program` is not a string
     KeyError
         `program` is not an executable dependency.
-
     """
-    dispatcher = {
-        'prodigal': prodigal,
-        'diamond': diamond,
-        'hmmsearch': hmmsearch,
-        'hmmpress': hmmpress,
-        'hmmscan': hmmscan,
-        'prodigal': prodigal,
-        'bowtie2': bowtie2,
-        'samtools': samtools,
-        'bedtools': bedtools,
-    }
     if program:
-        exe_name = os.path.basename(program)
         if type(program) is not str:
-            raise ValueError(f'program is not string. given:{type(program)}')
-        if exe_name not in dispatcher:
+            raise TypeError(f'program is not string. given:{type(program)}')
+        exe_name = os.path.basename(program)
+        if exe_name not in globals():
             raise KeyError(f'{exe_name} not in executables')
-        return dispatcher[exe_name]()
+        return (globals()[exe_name]())
     versions = {}
     executables = find_executables()
     for exe, found in executables.items():
         if found:
-            version = dispatcher[exe]()
+            # globals()[exe]() accesses the location of the function for that program
+            # eg. location of bedtools(), prodigal(), etc..
+            version = globals()[exe]()
```

Comment:
Maybe we should split this into two lines for readability? 

I'm thinking something like:

```suggestion
            get_version = globals()[exe]
            version = get_version()
```

Perhaps this would be easier to understand for the future reader?